### ISSUE TITLE
False positive error on the bevy Asset trait

### ISSUE THREAD

**Issue by redpandamonium** [intellij-rust/intellij-rust#8581]
## Environment

* **IntelliJ Rust plugin version: 0.4.165.4438-213**
* **Rust toolchain version: stable-x86_64-unknown-linux-gnu @ 1.57.0**
* **IDE name and version: CLion 2021.3.3**
* **Operating system: Manjaro Linux**

## Problem description
The code below compiles, but the plugin detects an error. 

## Steps to reproduce

Use bevy version 0.6. The following code compiles using the specified Rust toolchain. However in the IDE the marked part is highlighted as an error.

```Rust
use bevy::reflect::{TypeUuid, TypeUuidDynamic};
use bevy::asset::Asset;

trait DerivedAsset: Asset { }

#[derive(Clone, Default, TypeUuid)]
#[uuid = "45e22445-c6f6-455c-9527-1bf9932193e3"]
struct SomeReflectedType { }

impl DerivedAsset for SomeReflectedType { } //< the trait bound `SomeReflectedType: Asset` is not satisfied [E0277] the trait `Asset` is not implemented for `SomeReflectedType`

fn main() {
    println!("Hello, world!");
}
```

Sadly I'm not rust-savvy enough to understand the magic going on inside bevy here. So I can't reduce it further.

Here's a repository that reproduces this bug: https://github.com/redpandamonium/intellij_rust_bug_02

**Comment by Undin** [comment]
@redpandamonium Thanks for the report!

It's expected for now because the Rust plugin doesn't expand procedural macros by default. As a result, the plugin doesn't know about `bevy::reflect::TypeUuid` implementation for `DerivedAsset`.

But there is an experimental procedural macro expansion in the plugin. To turn it on, enable `org.rust.cargo.evaluate.build.scripts` and `org.rust.macros.proc` [experimental](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features) features. See #6908 for more details

**Comment by redpandamonium** [comment]
Thanks for your response! I can confirm that enabling these experimental features does indeed fix the bug.
